pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListener.java

```diff
@@ -41,52 +53,73 @@
  * @see <a href="https://github.com/apache/pulsar/wiki/PIP-82%3A-Tenant-and-namespace-level-rate-limiting">Global-quotas</a>
  *
  */
-public class ResourceGroupConfigListener implements Consumer<Notification> {
+public class ResourceGroupConfigListener implements Consumer<Notification>, AutoCloseable {
 
     private static final Logger LOG = LoggerFactory.getLogger(ResourceGroupConfigListener.class);
     private final ResourceGroupService rgService;
     private final PulsarService pulsarService;
     private final ResourceGroupResources rgResources;
-    private final ResourceGroupNamespaceConfigListener rgNamespaceConfigListener;
+    private volatile ResourceGroupNamespaceConfigListener rgNamespaceConfigListener;
+    private final ScheduledExecutorService executorService;
 
     public ResourceGroupConfigListener(ResourceGroupService rgService, PulsarService pulsarService) {
         this.rgService = rgService;
         this.pulsarService = pulsarService;
         this.rgResources = pulsarService.getPulsarResources().getResourcegroupResources();
-        loadAllResourceGroups();
         this.rgResources.getStore().registerListener(this);
-        rgNamespaceConfigListener = new ResourceGroupNamespaceConfigListener(
-                rgService, pulsarService, this);
+        this.executorService =
+                Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("pulsar-resource-group-config"));
+        execute(() -> loadAllResourceGroupsWithRetryAsync(0));
     }
 
-    private void loadAllResourceGroups() {
-        rgResources.listResourceGroupsAsync().whenCompleteAsync((rgList, ex) -> {
-            if (ex != null) {
-                LOG.error("Exception when fetching resource groups", ex);
-                return;
+    @SneakyThrows
+    private void loadAllResourceGroupsWithRetryAsync(long retry) {
+        loadAllResourceGroupsAsync().exceptionally(e -> {
+            long nextRetry = retry + 1;
+            long delay = 500 * nextRetry;
+            LOG.error("Failed to load all resource groups during initialization, retrying after {}ms", delay);
+            schedule(() -> loadAllResourceGroupsWithRetryAsync(nextRetry), delay);
+            throw new RuntimeException(e);
+        }).thenAccept(__ -> {
+            if (rgNamespaceConfigListener == null) {
+                try{
+                    rgNamespaceConfigListener = new ResourceGroupNamespaceConfigListener(rgService, pulsarService, this);
+                }catch (Exception e) {
+                    e.printStackTrace();
+                }
             }
+        });
+    }
+
+    private CompletableFuture<Void> loadAllResourceGroupsAsync() {
+        return rgResources.listResourceGroupsAsync().thenComposeAsync(rgList -> {
```

Comment:
we can use the executor for this async

Reply:
`executorService` only is used to retry the `loadAllResourceGroupsAsync()`.